### PR TITLE
Ftr: add BigInteger support

### PR DESCRIPTION
**Pull request by micln**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test_hessian/src/main/java/test/TestCustomDecode.java [hunk 1]
```diff
@@ -166,6 +167,11 @@ public Object customArgTypedFixedList_Object() throws Exception {
         return Arrays.deepEquals(list, (Object[]) o);
     }
 
+    public Object customArgTypedFixedList_Integer() throws Exception {
```
Comment (wongoo) [inline test_hessian/src/main/java/test/TestCustomDecode.java]: the method name should remove `TypedFixedList` ,i think you can change other method name customArgTypedFixedList_Decimal together

File: serialize_test.go [hunk 2]
```diff
@@ -69,3 +69,46 @@ func doTestDecimal(t *testing.T, method, content string) {
...
+}
+
+func TestIntegerJavaDecode(t *testing.T) {
+	var d big.Integer
+	_ = d.FromString("100256")
+	d.Value = d.String()
+	testJavaDecode(t, "customArgTypedFixedList_Integer", d)
+}
+
+func doTestInteger(t *testing.T, method, content string) {
```
Comment (wongoo) [inline serialize_test.go]: not need to create new method for only used in one place

File: test_hessian/src/main/java/test/TestCustomDecode.java (hunk 1 above)
Comment (micln) [reply test_hessian/src/main/java/test/TestCustomDecode.java]: both `customArgTypedFixedList_Decimal ` and `customArgTypedFixedList_DateNull` fixed together

File: date_test.go [hunk 3]
```diff
@@ -20,9 +20,7 @@ package hessian
 import (
 	"testing"
 	"time"
-)
 
-import (
```
Comment (micln) [inline date_test.go]: my IDE autoformat plugins merge this to one group, if it make something bad, i can rollback it

**Comment by wongoo** [comment]
this pr depends on the pr of gost, which must be merged and release a new version first.

File: go.mod [hunk 4]
```diff
@@ -1,7 +1,11 @@
 module github.com/apache/dubbo-go-hessian2
 
+replace github.com/dubbogo/gost => github.com/micln/gost v1.3.1-0.20191128163547-b75faeb257b0
```
Comment (micln) [inline go.mod]: - [ ] need remove

File: serialize_test.go (hunk 2 above)
Comment (micln) [reply serialize_test.go]: merged into doTestStringer

**Comment by wongoo** [comment]
LGTM

**Comment by wongoo** [comment]
@micln  do u have more unit tests to commit? I'd like to merge it if nothing.

**Comment by micln** [comment]
@wongoo sorry I will finish it tonight

**Comment by wongoo** [comment]
@micln  the bug that can't decoding BigInteger array  fixed by  https://github.com/apache/dubbo-go-hessian2/pull/144

File: serialize.go [hunk 5]
```diff
@@ -42,6 +47,32 @@ func GetSerializer(key string) (Serializer, bool) {
...
+
+func (IntegerSerializer) DecObject(d *Decoder) (interface{}, error) {
+	bigInt, err := d.DecodeValue()
+	if err != nil {
+		return nil, err
+	}
+
+	result, ok := bigInt.(*bigInteger)
+	if !ok {
+		panic("result type is not Integer, please check the whether the conversion is ok")
```
Comment (micln) [inline serialize.go]: @wongoo how about return this to err

Comment (wongoo) [reply serialize.go]: yes，but i suggest do that in another PR.

File: serialize.go [hunk 6]
```diff
@@ -18,18 +18,26 @@
 package hessian
 
 import (
+	"log"
 	"reflect"
-)
```
Comment (wongoo) [inline serialize.go]: split the imports

File: date_test.go [hunk 7]
```diff
@@ -20,9 +20,7 @@ package hessian
 import (
 	"testing"
 	"time"
-)
```
Comment (wongoo) [inline date_test.go]: split the imports

File: serialize.go [hunk 8]
```diff
@@ -18,18 +18,26 @@
...
+
 func init() {
+	RegisterPOJO(&bigInteger{})
+	SetSerializer("java.math.BigInteger", IntegerSerializer{})
+
 	RegisterPOJO(&big.Decimal{})
 	SetSerializer("java.math.BigDecimal", DecimalSerializer{})
 }
 
+func init() {
```
Comment (wongoo) [inline serialize.go]: remove the log init

File: serialize_test.go [hunk 9]
```diff
@@ -18,11 +18,10 @@
 package hessian
 
 import (
+	"fmt"
 	"reflect"
 	"testing"
-)
```
Comment (wongoo) [inline serialize_test.go]: split imports

**Pull request merged**